src/utility_belt/sql/helpers.clj

```diff
@@ -1,15 +1,31 @@
 (ns utility-belt.sql.helpers
-  (:require [next.jdbc]
-            [utility-belt.sql.model :as model]))
+  (:require
+    [next.jdbc]
+    [utility-belt.sql.model :as model]))
+
 
 (defmacro with-transaction
-  "Wrapper around jdbc transaction macro. Cleans up imports basically"
+  "Wrapper around jdbc transaction macro.   Cleans up imports so we always use ut.sql
+  Suports the same options, eg.
+  [tx conn]
+  and
+  [tx conn opts]
+  + body."
   [binding & body]
   `(next.jdbc/with-transaction ~binding ~@body))
 
+
 (defn execute
   ([connection statement]
    (execute connection statement {:mode :kebab-maps}))
   ([connection statement {:keys [mode]}]
    (next.jdbc/execute! connection statement
                        (get model/modes mode))))
+
+
+(defn transaction?
+  "Detects if passed in data source is a transaction or not.
+  NOTE: this only works with HikariCP managed connections/transactions and the connection pool
+  defined in utility-belt.sql.component.connection-pool"
+  [tx-or-conn]
+  (-> tx-or-conn bean :transactionIsolation number?))
```

Comment:
I measured this, takes ~2ms to check this so overhead is negligible

Reply:
@AfonsoTsukamoto to keep you in the loop. I have discussion with Lukasz about the transaction and he is right about moving them to handler layer. But in model layer we should always check that we passed as argument transaction (rather than db-pool) if required. With this we can start fixing lucien slowly rather than huge refactor.

Reply:
Thanks Marketa. I saw the comments in the other PR an also found this https://github.com/seancorfield/next-jdbc/blob/91dda2cdae3f9e897a54fe21edf8467acae8aa0d/src/next/jdbc/transaction.clj#L40
It is set per thread by next.jdbc when a transaction starts so that they can also detect if a transaction is nested. I guess both solutions would work.